### PR TITLE
 Refactor

### PR DESCRIPTION
**Pull request by tuannvm**
https://trello.com/c/Rb9BO53q/872-dex-app-fixes

## Why?
- Refactor

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -282,6 +282,8 @@ func (a *app) handleLogin(w http.ResponseWriter, r *http.Request) {
 func (a *app) handleDownload(w http.ResponseWriter, r *http.Request) {
 	var refreshToken = r.FormValue("refresh_token")
 	var idToken = r.FormValue("id_token")
+	var internal = r.FormValue("internal")
```
Comment (tuannvm) [inline main.go]: Add support for internal endpoint.

File: main.go [hunk 2]
```diff
@@ -361,5 +368,5 @@ func (a *app) handleCallback(w http.ResponseWriter, r *http.Request) {
 	buff := new(bytes.Buffer)
 	json.Indent(buff, []byte(claims), "", "  ")
 
-	renderToken(w, ClientClusters[ClientID]["CACert"], ClientID, ClientClusters[ClientID]["ClusterEndpoint"], a.redirectURI, rawIDToken, token.RefreshToken, buff.Bytes())
+	renderToken(w, ClientClusters[ClientID]["CACert"], ClientID, ClientClusters[ClientID]["ClusterEndpoint"], a.redirectURI, rawIDToken, token.RefreshToken, buff.Bytes(), ClientNamespaces)
```
Comment (tuannvm) [inline main.go]: Render namespace

**Comment by so0k** [comment]
don't think we need a full theme.. didn't see PR until now - just approved so merge and deploy

File: main.go [hunk 3]
```diff
@@ -299,7 +303,7 @@ func (a *app) handleDownload(w http.ResponseWriter, r *http.Request) {
 	} else {
 		clusterEndpoint = ClientClusters[ClientID]["ClusterEndpoint"]
 	}
-	renderKubeConfig(w, ClientID, ClientClusters[ClientID]["CACert"], clusterEndpoint, idToken, refreshToken)
+	renderKubeConfig(w, ClientID, ClientClusters[ClientID]["CACert"], clusterEndpoint, idToken, refreshToken, namespace)
```
Comment (tuannvm) [inline main.go]: Render kubeconfig with namespace

File: main.go [hunk 4]
```diff
@@ -44,6 +44,9 @@ var ClientClusters = map[string]map[string]string{
 	},
 }
 
+// ClientNamespaces : predefined slice of namespaces
+var ClientNamespaces = strings.Split(os.Getenv("NAMESPACES"), ",")
```
Comment (tuannvm) [inline main.go]: get namespaces (comma-separated) from `env`

**Pull request merged**